File: AN_2/SEM_1/MAP/lab2/src/tests/PrinterTaskRunnerTest.java
package tests;

import model.MessageTask;
import decorator.PrinterTaskRunner;
import decorator.StrategyTaskRunner;
import Enum.Strategy;

public class PrinterTaskRunnerTest
{
    public static void printerTaskRunnerTest(Strategy strategy, MessageTask[] messageTasks)
    {
        StrategyTaskRunner strategyTaskRunner = new StrategyTaskRunner(strategy);
        PrinterTaskRunner printerTaskRunner = new PrinterTaskRunner(strategyTaskRunner);
        printerTaskRunner.addTask(messageTasks[0]);
        printerTaskRunner.addTask(messageTasks[1]);
        printerTaskRunner.addTask(messageTasks[2]);
        printerTaskRunner.executeAll();
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Tests/PrinterTaskRunnerTest.cs
using Model;
using Decorator;
using Enum;

namespace Tests
{
    public class PrinterTaskRunnerTest
    {
        public static void RunPrinterTaskRunnerTest(Strategy strategy, MessageTask[] messageTasks)
        {
            var strategyTaskRunner = new StrategyTaskRunner(strategy);
            var printerTaskRunner = new PrinterTaskRunner(strategyTaskRunner);
            printerTaskRunner.AddTask(messageTasks[0]);
            printerTaskRunner.AddTask(messageTasks[1]);
            printerTaskRunner.AddTask(messageTasks[2]);
            printerTaskRunner.ExecuteAll();
        }
    }
}
